composer/workflows/dataflowtemplateoperator_create_dataset_and_table_test.py

```diff
@@ -0,0 +1,48 @@
+from google.cloud import bigquery
+import google.cloud.exceptions
+import pytest
+from . import dataflowtemplateoperator_create_dataset_and_table_helper as helper
+
+client = bigquery.Client()
+
+expected_schema = schema = [
+    bigquery.SchemaField("location", "GEOGRAPHY", mode="REQUIRED"),
+    bigquery.SchemaField("average_temperature", "INTEGER", mode="REQUIRED"),
+    bigquery.SchemaField("month", "STRING", mode="REQUIRED"),
+    bigquery.SchemaField("inches_of_rain", "NUMERIC", mode="NULLABLE"),
+    bigquery.SchemaField("is_current", "BOOLEAN", mode="NULLABLE"),
+    bigquery.SchemaField("latest_measurement", "DATE", mode="NULLABLE"),
+]
+
+expected_table_id = f"{client.project}.average_weather.average_weather"
+expected_dataset_id = f"{client.project}.average_weather"
+
+
+@pytest.fixture(scope="module")
+def dataset():
+    try:
+        dataset = client.get_dataset(expected_dataset_id)
+    except google.cloud.exceptions.NotFound:
+        dataset = helper.create_dataset()
+
+    yield dataset
+
+    client.delete_dataset(dataset, delete_contents=True, not_found_ok=True)
+
+
+@pytest.fixture(scope="module")
+def table():
+    try:
+        table = client.get_table(expected_table_id)
+    except google.cloud.exceptions.NotFound:
+        table = helper.create_table()
+
+    yield table
+
+    client.delete_table(table, not_found_ok=True)
+
+
+def test_creation(dataset, table):
```

Comment:
Well, I think a test should have invocation of a target function (a function we want to test) in the test body. Normally I write a test per target function. This case is simple, so I'm ok with just having one test for 2 functions.

My suggestion:

```python
@pytest.fixture(scope="module")
def cleaner():
    datasets_to_delete = []

    yield datasets_to_delete

    for dataset in datasets_to_delete:
        client.delete_dataset(dataset, delete_contents=True, not_found_ok=True)


def test_create_dataset_and_table(capsys, cleaner):
    helper.create_dataset(project_id, dataset_id)
    # Immediately add the dataset for cleaning up.
    cleaner.append(dataset_id)
    created_table = helper.create_table(....)

    assert expected_schema == created_table.schema. # I'm not sure it works.
    # then also assert the output ...
```